rust/otap-dataflow/crates/otap/src/filter_processor/filter_logs.rs

```diff
@@ -0,0 +1,381 @@
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#displaying-severity
+	// this field is case-insensitive ("INFO" == "info")
+	min: i32,
+	// MatchUndefined lets logs records with "unknown" severity match.
+	// If MinSeverity is not set, this field is ignored, as fields are not matched based on severity.
+	match_undefined: bool,
+}
+
+
+pub enum LogMatchType {
+    Strict, 
+    Regexp
+}
+
+// ToDo if no filter is set then the filter must allow everything through, boolean array should be initialized to all true
+
+impl LogFilter {
+	pub fn filter(&self, logs_payload: OtapArrowRecord) -> Result<OtapArrowRecords, Error>{
+		let (include_resource_attr_filter, include_log_record_filter, include_log_attr_filter) = self.include.create_filters(logs_payload, false)?;
+		let (exclude_resource_attr_filter, exclude_log_record_filter, exclude_log_attr_filter) = self.exclude.create_filters(logs_payload, true)?;
+
+		// combine the include and exclude filters
+		let resource_attr_filter = arrow::compute::or_kleene(&include_resource_attr_filter, &exclude_resource_attr_filter).map_err()?;
+		let log_record_filter = arrow::compute::or_kleene(&include_log_record_filter, &exclude_log_record_filter).map_err()?;
+		let log_attr_filter = arrow::compute::or_kleene(&include_log_attr_filter, &exclude_log_attr_filter).map_err()?;
+
+		let resource_attrs = logs_payload.get(ArrowPayLoadType::ResourceAttrs)?;
+		let log_records = logs_payload.get(ArrowPayLoadType::Logs)?;
+		let log_attrs = logs_payload.get(ArrowPayLoadType::LogAttrs)?;
+
+
+        // HERE WE CLEAN UP THE TABLE WE EXTRACT THE IDS OF RECORDS WE ARE REMOVING AND MAKE SURE TO SYNC THESE REMOVALS WITH THE OTHER TABLES
+
+        // we start with the resource_attr we get the id of the resource_attr that are being removed and use that to update the log_records filter
+        // we want to remove the log_records the matching parent_id as the resource_attr that have been removed
+
+
+        // create filters that we will use to get the rows that are getting removed
+        // use these filters to get the ids of rows that are getting removed from the resource_attr
+        let id_resource_attr_filter = arrow::compute::not(&resource_attr_filter).map_err()?;
+        let resource_id_column = resource_attrs.column_by_name("parent_id").map_err()?;
+        // we have a have a array contain all the parent_ids that will be removed
+
+        // get ids that we being kept
+        let resource_ids_filtered = arrow::compute::filter(&resource_id_column, resource_attr_filter).map_err()?;
+        // get ids being removed
+        let resource_ids_removed = arrow::compute::filter(&resource_id_column, id_resource_attr_filter).map_err()?;
+
+        // update the log_records_filter to remove records that contain the parent_ids we found above
+        let log_record_resource_id_column = log_records.column_by_name("resource_id").map_err()?;
```

Comment:
I think the resource ID column is actually embedded within a `StructArray` on the log record. It can be accessed like:
```rs
        let struct_array = get_required_array(log_records, consts::RESOURCE)?;
        let struct_array = struct_array
            .as_any()
            .downcast_ref::<StructArray>()
            .with_context(|| error::ColumnDataTypeMismatchSnafu {
                name: consts::RESOURCE,
                actual: struct_array.data_type().clone(),
                expect: Self::data_type().clone(),
            })?;
```